SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+                $NewTempVolume = $NewTempVolume.Substring(0, $NewTempVolume.Length - 1)
+                $TempVolume = $NewTempVolume
+                Write-Host "TempVolume moved to " $TempVolume
+            }
+            else {
+                $TempVolume = $NewTempVolume
+                Write-Host "TempVolume moved to " $TempVolume
+            }
+            If ([string]::IsNullOrEmpty($NewTempVolume)) {
+                Write-Host "TempVolume remains on " $TempVolume
+            }
+
+            Write-Host "AppVolume: " $AppVolume
+            $NewAppVolume = Read-Host "Your AppVolume: "
+            If ($NewAppVolume.Substring($NewAppVolume.Length - 2 -eq $CheckLastTwoChar) -and $NewAppVolume.Length -gt 2) {
+                $NewAppVolume = $NewAppVolume.Substring(0, $NewAppVolume.Length - 2)
+                $AppVolume = $NewAppVolume
+                Write-Host "AppVolume moved to " $AppVolume
+            }
+            elseif ($NewAppVolume.Substring($NewAppVolume.Length - 1 -eq $CheckLastChar) -and $NewAppVolume.Length -gt 1) {
+                $NewAppVolume = $NewAppVolume.Substring(0, $NewAppVolume.Length - 1)
+                $AppVolume = $NewAppVolume
+                Write-Host "AppVolume moved to " $AppVolume
+            }
+            else {
+                $AppVolume = $NewAppVolume
+                Write-Host "AppVolume moved to " $AppVolume
+            }
+            If ([string]::IsNullOrEmpty($NewAppVolume)) {
+                Write-Host "AppVolume remains on " $AppVolume
+            }
+
+            Write-Host "BackupVolume: " $BackupVolume
+            $NewBackupVolume = Read-Host "Your BackupVolume: "
+            If ($NewBackupVolume.Substring($NewBackupVolume.Length - 2 -eq $CheckLastTwoChar) -and $NewBackupVolume.Length -gt 2) {
+                $NewBackupVolume = $NewBackupVolume.Substring(0, $NewBackupVolume.Length - 2)
+                $BackupVolume = $NewBackupVolume
+                Write-Host "BackupVolume moved to " $BackupVolume
+            }
+            elseif ($NewBackupVolume.Substring($NewBackupVolume.Length - 1 -eq $CheckLastChar) -and $NewBackupVolume.Length -gt -1) {
+                $NewBackupVolume = $NewBackupVolume.Substring(0, $NewBackupVolume.Length - 1)
+                $BackupVolume = $NewBackupVolume
+                Write-Host "BackupVolume moved to " $BackupVolume
+            }
+            else {
+                $BackupVolume = $NewBackupVolume
+                Write-Host "BackupVolume moved to " $BackupVolume
+            }
+            If ([string]::IsNullOrEmpty($NewBackupVolume)) {
+                Write-Host "BackupVolume remains on " $BackupVolume
+            }
+        }
+        Default {Write-Host "Drives agreed, continuing"; }
+    }
+
+    $CheckLastTwoChar = ":\"
+    $CheckLastChar = "\"
+
+    $SetupFile = Read-Host -Prompt 'Please enter the root location for Setup.exe'
```

Comment:
This would be a required command parameter as opposed to a prompt (tho I'd likely try to determine the path for them. Detect removable drives, and search for the path to sql server setup)